pandas/core/arrays/categorical.py

```diff
@@ -523,6 +527,11 @@ def _from_inferred_categories(cls, inferred_categories, inferred_codes,
                 cats = to_datetime(inferred_categories, errors='coerce')
             elif is_timedelta64_dtype(dtype.categories):
                 cats = to_timedelta(inferred_categories, errors='coerce')
+            elif dtype.categories.is_boolean():
+                if true_values is None:
+                    true_values = _true_values
```

Comment:
you could *maybe* enhance ``to_numeric`` to take a ``downcasts='bool'`` to do this?